platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/WiringModel.java

```diff
@@ -0,0 +1,151 @@
+     * factors.
+     *
+     * @param period the period of the heartbeat. For example, setting a period of 100ms will cause the heartbeat to be
+     *               sent at 10 hertz. Note that time is measured at millisecond precision, and so periods less than 1ms
+     *               are not supported.
+     * @return the output wire
+     * @throws IllegalStateException if the heartbeat has already started
+     */
+    @NonNull
+    OutputWire<Instant> buildHeartbeatWire(@NonNull final Duration period);
+
+    /**
+     * Build a wire that produces an instant (reflecting current time) at the specified rate. Note that the exact rate
+     * of heartbeats may vary. This is a best effort algorithm, and actual rates may vary depending on a variety of
+     * factors.
+     *
+     * @param frequency the frequency of the heartbeat in hertz. Note that time is measured at millisecond precision,
+     *                  and so frequencies greater than 1000hz are not supported.
+     * @return the output wire
+     */
+    OutputWire<Instant> buildHeartbeatWire(final double frequency);
+
+    /**
+     * Check to see if there is cyclic backpressure in the wiring model. Cyclical back pressure can lead to deadlocks,
+     * and so it should be avoided at all costs.
+     *
+     * <p>
+     * If this method finds cyclical backpressure, it will log a message that will fail standard platform tests.
+     *
+     * @return true if there is cyclical backpressure, false otherwise
+     */
+    boolean checkForCyclicalBackpressure();
+
+    /**
+     * Task schedulers using the {@link TaskSchedulerType#DIRECT} strategy have very strict rules about how data can be
+     * added to input wires. This method checks to see if these rules are being followed.
+     *
+     * <p>
+     * If this method finds illegal direct scheduler usage, it will log a message that will fail standard platform
+     * tests.
+     *
+     * @return true if there is illegal direct scheduler usage, false otherwise
+     */
+    boolean checkForIllegalDirectSchedulerUsage();
+
+    /**
+     * Check to see if there are any input wires that are unbound.
+     *
+     * <p>
+     * If this method detects unbound input wires in the model, it will log a message that will fail standard platform
+     * tests.
+     *
+     * @return true if there are unbound input wires, false otherwise
+     */
+    boolean checkForUnboundInputWires();
+
+    /**
+     * Generate a mermaid style wiring diagram.
+     *
+     * @param groups optional groupings of vertices
+     * @return a mermaid style wiring diagram
+     */
+    @NonNull
+    String generateWiringDiagram(@NonNull final Set<ModelGroup> groups);
+
+    /**
+     * Start everything in the model that needs to be started. Performs static analysis of the wiring topology and
+     * writes errors to the logs if problems are detected.
+     */
+    @Override
+    void start();
```

Comment:
`start` and `stop` definitions here are redundant, since they are already required on implementing classes via `Startable` and `Stoppable`

Reply:
The only reason why I include them is so that I can add javadocs that are specific to the wiring model (as opposed to the generic ones on the parent interface).

```
    /**
     * Start everything in the model that needs to be started. Performs static analysis of the wiring topology and
     * writes errors to the logs if problems are detected.
     */
    @Override
    void start();
```